sys/net/gnrc/transport_layer/udp/gnrc_udp.c

```diff
@@ -130,9 +130,9 @@ static void _receive(gnrc_pktsnip_t *pkt)
     port = (uint32_t)byteorder_ntohs(hdr->dst_port);
 
     /* send payload to receivers */
-    if (!gnrc_netapi_dispatch_receive(GNRC_NETTYPE_UDP, port, pkt)) {
+    if (!gnrc_netapi_dispatch_receive(GNRC_NETTYPE_UDP, port, udp)) {
```

Comment:
What no! The listener is *supposed* to get the whole packet, not just the payload. And you are not sending the payload, you are sending the UDP header... at least when the meaning of the variables is still the same as it was.